tests/rbd/test_rbd_compression.py

```diff
@@ -0,0 +1,222 @@
+    data is getting compressed or not.
+
+    Args:
+        mode: type of compression hints (compressible or incompressible)
+        image_spec: poolname + imagename
+        **kw: test data
+
+    Returns:
+        int: The return value. 0 for success, 1 for failure.
+    """
+    config = kw.get("config")
+    io = config.get("io_total")
+    kw["io"] = io
+    kw["imagespec"] = image_spec
+    pool_name = image_spec.split("/")[0]
+
+    # running io to check compression of data
+    self.rbd_bench(**kw)
+
+    out = self.exec_cmd(cmd="ceph df detail --format json-pretty", output=True)
+    output = json.loads(out.rstrip())
+    pool_info = output["pools"]
+
+    # Iterate over the pools to get compressed data information
+    for pool in pool_info:
+        if pool["name"] == pool_name:
+            compressed_data = pool["stats"]["compress_bytes_used"]
+            break
+    log.debug(pool["stats"])
+    if mode == "compressible":
+        if compressed_data == 0:
+            log.error("Test Failed: Data did not get compressed in compressible mode")
+            return 1
+        else:
+            log.info(
+                f"Test Passed: Data got compressed, bytes compressed: {compressed_data}"
+            )
+            return 0
+
+    elif mode == "incompressible":
+        if compressed_data != 0:
+            log.error("Test Failed: Data gets compressed in incompressible mode")
+            return 1
+        else:
+            log.info(
+                "Test Passed: Data did not get compressed due to incompressible mode set"
+            )
+            return 0
+    else:
+        log.error(f"Invalid mode: {mode}")
+        return 1
+
+
+def test_compression(rbd, pool_type, **kw):
+    """
+    Run rbd_compression_hint test on supported options,
+    i.e compressible and incompressible.
+
+    Args:
+        rbd: RBD object
+        pool_type: pool type (ec_pool_config or rep_pool_config)
+        **kw: test data
+
+    Returns:
+        int: The return value. 0 for success, 1 for failure.
+    """
+    config = kw.get("config")
+
+    compression_algorithm = config.get("compression_algorithm")
+    compression_mode = config.get("compression_mode")
+    compression_ratio = config.get("compression_ratio")
+
+    pool_name = config[pool_type]["pool"]
+
+    compression_configs = {
+        "compression_algorithm": compression_algorithm,
+        "compression_mode": compression_mode,
+        "compression_required_ratio": compression_ratio,
+    }
+
+    try:
+        # set bluestore_compression_mode to passive
+        rbd.set_ceph_config("osd", "bluestore_compression_mode", "passive")
+
+        if (
+            rbd.get_ceph_config(
+                "osd", "bluestore_compression_mode", output=True
+            ).strip()
+            != "passive"
+        ):
+            log.error("Not able to set bluestore_compression_mode to passive")
+            return 1
+
+        log.info("Able to set bluestore_compression_mode to passive")
```

Comment:
The sentence could be "bluestore_compression_mode set to passive"

Reply:
modified accordingly